resources/test/mock-ad-units.js

```diff
@@ -1,63 +1,269 @@
-  },
-
-  units: {
-
-    'header': {
-      slotName: 'header',
-      sizes: [
-        [[970, 0], [[728, 90], [970, 250], [970, 90], [970, 415]]],
-        [[728, 0], [728, 90]],
-        [[0, 0], [320, 50]]
-      ]
-    },
-
-    'article-header': {
-      slotName: 'header',
-      sizes: [
-        [[970, 0], [[728, 90], [970, 250]]],
-        [[728, 0], [728, 90]],
-        [[0, 0], [320, 50]]
-      ]
-    },
-
-    'sidebar-primary': {
-      sizes: [
-        [[0, 0], [300, 250]]
-      ]
-    },
-
-    'sidebar-secondary': {
-      sizes: [
-        [[0, 0], [300, 250]]
-      ]
-    },
-
-    'sidebar-tertiary': {
-      sizes: [
-        [[0, 0], [300, 250]]
-      ]
+    closeDelay: 1250
+  },
+
+  makeAdCloseable: function(adElement) {
+    var closeButton = document.createElement('div');
+    if (this.settings.closeDelay) {
+      closeButton.className = 'close-btn disabled';
+      setTimeout(function() {
+        utils.removeClass(closeButton, 'disabled');
+      }, this.settings.closeDelay);
+    } else {
+      closeButton.className = 'close-btn';
+    }
+    utils.addClass(adElement.parentElement, 'pinned');
+    adElement.appendChild(closeButton);
+
+    if('ontouchend' in document.documentElement) {
+      closeButton.addEventListener('touchend', AdUnits.closeAd.bind(null, adElement), false);
+    } else {
+      closeButton.addEventListener('click', AdUnits.closeAd.bind(null, adElement), false);
+    }
+  },
+
+  closeAd: function(adElement) {
+    if (onionan) {
+      onionan.sendEvent({
+        eventCategory: 'ads',
+        eventAction: 'close',
+        eventLabel: ''
+      });
+    } else {
+      console.warn('onionan is undefined');
+    }
+    adElement.parentElement.removeChild(adElement);
+  },
+
+  prependSite: function(baseSlotName) {
+    return this.settings.dfpSite + '_' + baseSlotName;
+  },
+
+  resetClasses: function(parent) {
+    utils.removeClass(parent, 'pinned');
+    utils.removeClass(parent, 'mobile');
+    utils.removeClass(parent, 'kargo');
+  },
+
+  isKargo: function(e) {
+    var re = new RegExp(/kargo/);
+    var slot = e.slot.L;
+    if (re.test(slot) || e.lineItemId === 356279568 || e.lineItemId === 336705048) {
+      return true;
+    } else {
+      return false;
+    }
+  },
+
+  setupMobileSlotClasses: function(e, el) {
+    var parent = el.parentElement;
+    utils.addClass(parent, 'mobile');
+
+    if (AdUnits.isKargo(e)) {
+      utils.addClass(parent, 'kargo');
     }
+  },
+
+  handleMobileHeaderSlot: function(e, el) {
+    var parent = el.parentElement;
+    AdUnits.makeAdCloseable(el);
+    if (!utils.hasClass(parent, 'header-wrapper')) {
+      return;
+    }
+
+    AdUnits.setupMobileSlotClasses(e, el);
+  },
+
+  handleLeaderboardHeaderSlot: function(e, el) {
+    AdUnits.makeAdCloseable(el);
+  },
+
+  headerSlotRenderEnded: function(e, el) {
+    var parent = el.parentElement;
+    AdUnits.resetClasses(parent);
+
+    if (e.size[0] === 320) {
```

Comment:
These numbers should be stored as constants somewhere. At the moment, it's difficult to tell what they refer to and where all their references are.